modules/client-common/src/main/java/org/apache/ignite/internal/client/table/MutableTupleBinaryTupleAdapter.java

```diff
@@ -256,6 +257,22 @@ public double doubleValue(int columnIndex) {
                 : binaryTuple.doubleValue(validateSchemaColumnType(columnIndex, ColumnType.DOUBLE));
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public BigDecimal decimalValue(String columnName) {
+        return tuple != null
+                ? tuple.decimalValue(columnName)
+                : binaryTuple.decimalValue(validateSchemaColumnType(columnName, ColumnType.DECIMAL), -1);
```

Comment:
based on javaDoc we should pass Integer. MIN_VALUE as scale for case when we going to retrieve scale as is